compiler/transforms/set_memory_layout.py

```diff
@@ -0,0 +1,99 @@
+from xdsl.dialects import builtin, linalg
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+from compiler.dialects.snax import LayoutCast
+from compiler.dialects.tsl import TiledStridedLayoutAttr
+from compiler.ir.tsl import Stride, TiledStride, TiledStridedLayout
+
+
+class AddMemoryLayout(RewritePattern):
+    """
+    This class represents a rewrite pattern for adding memory layout to a
+    linalg operation. The implementation is very naive. It imposes a specific
+    memory layout on the input and output of the linalg operation dispatched
+    to snax_qgemm by inserting reshuffling ops. In the future, the memory
+    layout will be selected in a more automatic way.
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, linalg_op: linalg.Generic, rewriter: PatternRewriter):
+        # check if operation is dispatched via library call
```

Comment:
```suggestion
        # check if operation is dispatched via library call, as set by e.g. the linalg-to-library-call pass
```

Reply:
The `linalg-to-library-call` does not set the library call? Maybe this is not a good name, but the pass transforms a `linalg` (with a library call) to an actual library call (insert `funcOps`)